packages/expo-updates/ios/EXUpdates/EXUpdatesModule.m

```diff
@@ -132,6 +138,40 @@ - (NSDictionary *)constantsToExport
   }];
 }
 
+EX_EXPORT_METHOD_AS(readLogEntriesAsync,
+                     readLogEntriesAsync:(NSNumber *)maxAge
+                                 resolve:(EXPromiseResolveBlock)resolve
+                                  reject:(EXPromiseRejectBlock)reject)
+{
+  // Once persistent logs are implemented, the iOS 15 restriction will be removed
+  if (@available(iOS 15.0, *)) {
+    EXUpdatesLogReader *reader = [EXUpdatesLogReader new];
+    NSError *error = nil;
+    NSInteger age = 3600; // default 1 hour
+    if (maxAge != nil) {
+      // maxAge is in milliseconds, convert to seconds to compute NSDate
+      age = [maxAge intValue] / 1000;
+    }
+    NSDate *epoch = [NSDate dateWithTimeIntervalSinceNow:-age];
+    NSArray<NSDictionary *> *entries = [reader getLogEntriesNewerThan:epoch error:&error];
+    if (error != nil) {
+      reject([NSString stringWithFormat:@"%ld",[error code]], [error localizedDescription], error);
```

Comment:
```suggestion
      reject([NSString stringWithFormat:@"%ld", error.code], error.localizedDescription, error);
```

Reply:
Replacing the code with the expected string `"ERR_UPDATES_READ_LOGS"`... I tend to use selectors instead of properties because they won't crash when called on a null object :)

Reply:
Dot notation is syntactic sugar for selector bracket notation so it has the same null target object semantics (I think). As far as I can tell, most of our objective-c uses dot syntax for property assignment and retrieval.